src/auth0-sso-login.js

```diff
@@ -6,13 +6,55 @@ export class windowInteraction {
   static updateWindow(url) {
     window.location = url;
   }
+
+  static setTimeout(func, delay) {
+    return window.setTimeout(func, delay);
+  }
+
+  static clearTimeout(timeoutId) {
+    window.clearTimeout(timeoutId);
+  }
 }
 
+// static localStorage interaction, mostly wrapped for enabling easier mocking through unit tests
+export class localStorageInteraction {
+  static getItem(key) {
+    return localStorage.getItem(key);
+  }
+
+  static setItem(key, value) {
+    return localStorage.setItem(key, value);
+  }
+
+  static removeItem(key) {
+    return localStorage.removeItem(key);
+  }
+}
+
+export const tokenExpiresAtKey = 'cimpress.auth0.tokenExpiresAt';
+
+const scheduleRenewal = (tokenRefreshHandle, renewFunction) => {
+  const tokenExpiresAtJson = localStorageInteraction.getItem(tokenExpiresAtKey);
+  if (!tokenExpiresAtJson) {
+    return null;
+  }
+  const tokenExpiresAt = JSON.parse(tokenExpiresAtJson);
```

Comment:
I'd move this to the `localStoreInteraction` let that instead be the `tokenExpiryManager`.  Either way the variable here probably should be instead `tokenExpiresAtString`